server/monitor-types/snmp.js

```diff
@@ -0,0 +1,66 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN, log, evaluateJsonQuery } = require("../../src/util");
+const snmp = require("net-snmp");
+
+class SNMPMonitorType extends MonitorType {
+    name = "snmp";
+
+    /**
+     * @inheritdoc
+     */
+    async check(monitor, heartbeat, _server) {
+
+        const options = {
+            port: monitor.port || "161",
+            retries: monitor.maxretries,
+            timeout: monitor.timeout * 1000,
+            version: snmp.Version[monitor.snmpVersion],
+        };
+
+        let session;
+        try {
+            session = snmp.createSession(monitor.hostname, monitor.radiusPassword, options);
```

Comment:
nit: Lets move this up a bit. Otherwise people get the idea to extract all options into a single variable at the start.

```suggestion
        let session;
        try {
            const sessionOptions = {
                port: monitor.port || "161",
                retries: monitor.maxretries,
                timeout: monitor.timeout * 1000,
                version: snmp.Version[monitor.snmpVersion],
            };
            session = snmp.createSession(monitor.hostname, monitor.radiusPassword, sessionOptions);
```